sam_api.c

```diff
@@ -744,23 +753,47 @@ void seader_iso14443a_transmit(
     SeaderWorker* seader_worker = seader->worker;
     SeaderCredential* credential = seader->credential;
 
-    BitBuffer* tx_buffer = bit_buffer_alloc(len);
+    BitBuffer* tx_buffer =
+        bit_buffer_alloc(len + 1); // extra byte to allow for appending a Le byte sometimes
     BitBuffer* rx_buffer = bit_buffer_alloc(SEADER_POLLER_MAX_BUFFER_SIZE);
 
     do {
-        if(credential->isDesfire && memcmp(buffer, ev2_request, len) == 0) {
-            FURI_LOG_I(TAG, "Intercept Desfire EV2 response and return File Not Found");
-            bit_buffer_append_bytes(rx_buffer, FILE_NOT_FOUND, sizeof(FILE_NOT_FOUND));
+        bit_buffer_append_bytes(tx_buffer, buffer, len);
 
-        } else {
-            bit_buffer_append_bytes(tx_buffer, buffer, len);
+        if(seader->credential->isDesfireEV2 && sizeof(select_desfire_app_no_le) == len &&
+           memcmp(buffer, select_desfire_app_no_le, len) == 0) {
+            // If a DESFire EV2 card has previously sent a dodgy reply to a SELECT SeosApp
```

Comment:
makes you wonder how the hell the readers do it

Reply:
Newer SAM firmware seems to avoid doing the SELECT SeosApp on DESFire cards in the first place, like I can replicate that via Seader even so it's not like we're just talking to the SAM wrong. The question is how it's identifying it, it doesn't have access to the ATQA, and oddly the ATS doesn't seem to matter for the most part either which really just leaves the fact it's a 14443a-4 card, the SAK, and the UID - none of which seems like it would be a reliable indicator, but perhaps the UID is more reliable than expected if you have full NXP datasheets or something.